source/renderer/app/components/wallet/WalletImporter.js

```diff
@@ -0,0 +1,255 @@
+    defaultMessage: '!!!Password',
+    description: 'Placeholder for the "Password" inputs in the wallet restore dialog.',
+  },
+});
+
+type Props = {};
+
+@observer
+export default class WalletImporter extends Component<Props> {
+
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  form = new ReactToolboxMobxForm({
+    fields: {
+      keyFile: {
+        label: this.context.intl.formatMessage(messages.keyFileLabel),
+        placeholder: this.context.intl.formatMessage(messages.keyFileHint),
+        type: 'file',
+      },
+      passwords: {
+        label: this.context.intl.formatMessage(messages.passwordsListLabel),
+        placeholder: this.context.intl.formatMessage(messages.passwordsListHint),
+        value: '',
+      },
+      walletHashes: {
+        value: [],
+      },
+      confirmedPasswordHashes: {
+        value: [],
+      },
+      spendingPassword: {
+        type: 'password',
+        label: this.context.intl.formatMessage(messages.spendingPasswordLabel),
+        placeholder: this.context.intl.formatMessage(messages.passwordFieldPlaceholder),
+        value: '',
+      },
+    },
+  }, {
+    options: {
+      validateOnChange: true,
+      validationDebounceWait: FORM_VALIDATION_DEBOUNCE_WAIT,
+    },
+  });
+
+  submit() {
+    const { form } = this;
+    const walletHashesField = form.$('walletHashes');
+    const spendingPasswordField = form.$('spendingPassword');
+    console.log(spendingPasswordField.value);
+    this.testPassword(spendingPasswordField.value);
+  }
+
+  checkPassword(passphraseHash, passhash) {
+    var bits = passphraseHash.split("|");
+    var logN = parseInt(bits[0]);
+    var r = parseInt(bits[1]);
+    var p = parseInt(bits[2]);
+    var salt = Buffer.from(bits[3],"base64");
+    var realhash = Buffer.from(bits[4],"base64");
+
+    // from nodejs native crypto (which is missing in renderer proc)
+    // var hash2 = scryptSync(cbor.encode(passhash), salt, 32, { N: 2 ** logN, r: r, p: p });
+    // try the scryptsy from npm (works, but worse performance)
+    var hash2 = scrypt(cbor.encode(passhash), salt, 2 ** logN, r, p, 32);
+    return realhash.equals(hash2);
+  }
+
+  testPassword(password) {
+    const { form } = this;
+    const walletHashesField = form.$('walletHashes');
+    const confirmedPasswordHashesField = form.$('confirmedPasswordHashes');
+
+    var testpasshash = null;
+    if (password == "") testpasshash = Buffer.from([]);
+    else testpasshash = Buffer.from(encryptPassphrase(password),"hex");
+
+    for (var idx=0; idx < walletHashesField.value.length; idx++) {
+      var match = this.checkPassword(walletHashesField.value[idx].pwhash, testpasshash);
+      if (match) {
+        console.log("wallet idx",idx,"has password", password);
+        var oldhashes = confirmedPasswordHashesField.value;
+        oldhashes[idx] = testpasshash.toString("hex");
+        confirmedPasswordHashesField.set(oldhashes);
+      }
+    }
+  }
+  extractKey(idx) {
```

Comment:
this creates the raw bytes that belong in a new `secret.key` file, that is destined to be imported by cardano
something will need to write these bytes to a file, call the import api, then delete the file